src/core/public/keyboard_shortcut/sequence_handler.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SEQUENCE_PREFIX } from './constants';
+import { SINGLE_LETTER_REGEX } from './utils';
+
+export class SequenceHandler {
+  private firstKey: string | null = null;
+  private sequenceTimer: number | null = null;
+
+  private readonly timeout: number;
+
+  constructor(timeout: number = 1000) {
+    this.timeout = timeout;
+  }
+
+  /**
+   * Normalizes and validates sequence key strings
+   *
+   * @param keyString - Sequence string to normalize and validate
+   * @returns Normalized sequence string
+   * @throws Error if sequence format is invalid
+   */
+  public normalizeKeyString(keyString: string): string {
+    const normalized = keyString.toLowerCase().trim();
+    const parts = normalized.split(/\s+/).filter((part) => part.length > 0);
```

Comment:
```suggestion
    const parts = normalized.split(' ').filter((part) => !!part.length);
```

Regex should only be used if standard string does not suffice, the problem with regex its hard to read